### PR TITLE
Adds support for custom dimensions to MUXSDKStats

### PR DESCRIPTION
**Pull request by nidhik**
## Zube
https://zube.io/muxinc/mux/c/18768

## Overview

Custom dimensions can now be set on monitor or on videoChange just like other types of customer metadata.

## Dependencies
* https://github.com/muxinc/mux-stats-sdk-objc/pull/117
* https://github.com/muxinc/stats-sdk-objc/pull/67

## QA
Check out the custom dimensions in the [views generated by our UI tests](https://dashboard.mux.com/organizations/ffhei0/environments/k0vjj8/views). 😄   

* Custom dimension `c1` now tags the test case that the view was generated for which makes it easy to filter for all the views that contained ads vs views for normal playback (for example).
* Custom dimension `c2` is overridden by the test case that updates custom dimensions. 

Screen Shot 2021-08-21 at 1 27 58 AM
Screen Shot 2021-08-21 at 1 28 14 AM
Screen Shot 2021-08-21 at 1 28 23 AM
Screen Shot 2021-08-21 at 1 29 09 AM

### CODE REVIEW

File: MUXSDKStats/MUXSDKStats/MUXSDKStats.m [hunk 1]
```diff
@@ -397,7 +411,15 @@ + (void)destroyPlayer:(NSString *)name {
...
++ (void)videoChangeForPlayer:(nonnull NSString *)name withCustomerData:(nullable MUXSDKCustomerData *)customerData {
+    MUXSDKCustomerPlayerData *playerData = [customerData customerPlayerData];
+    if (!playerData) {
+        playerData = [_customerPlayerDataStore playerDataForPlayerName:name];
+    }
+    MUXSDKCustomerViewData *viewData = [customerData customerViewData];
+    MUXSDKCustomerVideoData *videoData = [customerData customerVideoData];
+    MUXSDKCustomData *customData = [customerData customData];
+    
     if (!(videoData || viewData)) {
```
Comment (bmulligan-kg) [inline MUXSDKStats/MUXSDKStats/MUXSDKStats.m:423]: need to add `|| customData` here

File: MUXSDKStats/MUXSDKStats/MUXSDKStats.m [hunk 2]
```diff
@@ -451,6 +483,7 @@ + (void)setCustomerData:(nullable MUXSDKCustomerData *)customerData forPlayer:(n
     MUXSDKCustomerPlayerData *playerData = [customerData customerPlayerData];
     MUXSDKCustomerViewData *viewData = [customerData customerViewData];
     MUXSDKCustomerVideoData *videoData = [customerData customerVideoData];
+    MUXSDKCustomData *customData = [customerData customData];
 
     if (!playerData && !videoData && !viewData) return;
```
Comment (bmulligan-kg) [inline MUXSDKStats/MUXSDKStats/MUXSDKStats.m:488]: and here

Comment (bmulligan-kg) [reply MUXSDKStats/MUXSDKStats/MUXSDKStats.m:488]: i've actually flipped this around in my other PR, there will be a fair amount of conflicts between these two 😩

File: MUXSDKStats/MUXSDKStats/MUXSDKStats.m (hunk 1 above)
Comment (nidhik) [reply MUXSDKStats/MUXSDKStats/MUXSDKStats.m:423]: So...this is a draft PR lol (I'm still working on it)

But this comment just saved me! I was trying to figure out why one of my tests cases was not working for the last hour and this is the reason 🙏

File: MUXSDKStats/MUXSDKStats/MUXSDKStats.m (hunk 2 above)
Comment (nidhik) [reply MUXSDKStats/MUXSDKStats/MUXSDKStats.m:488]: :(

File: MUXSDKStats/MUXSDKStats/MUXSDKStats.m (hunk 1 above)
Comment (nidhik) [reply MUXSDKStats/MUXSDKStats/MUXSDKStats.m]: okay now this is ready

File: apps/DemoApp/DemoApp/ViewController.m [hunk 3]
```diff
@@ -99,7 +85,7 @@ - (void)viewWillTransitionToSize:(CGSize)size
        withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator {
     [coordinator animateAlongsideTransition:^(id<UIViewControllerTransitionCoordinatorContext> context) {} completion:^(id<UIViewControllerTransitionCoordinatorContext> context) {
         [MUXSDKStats orientationChangeForPlayer:DEMO_PLAYER_NAME withOrientation:[self viewOrientationForSize:size]];
-
+
```
Comment (skidder) [inline apps/DemoApp/DemoApp/ViewController.m:88]: Unnecessary whitespace

File: apps/DemoApp/DemoApp/ViewController.m [hunk 4]
```diff
@@ -207,13 +243,19 @@ - (void)changeProgram:(NSTimer *)timer {
     videoData.videoTitle = @"Apple Keynote";
     videoData.videoId = @"applekeynote2010";
     [MUXSDKStats programChangeForPlayer:DEMO_PLAYER_NAME
-                        withVideoData:videoData];
+                          withVideoData:videoData];
```
Comment (skidder) [inline apps/DemoApp/DemoApp/ViewController.m:246]: Whitespace

**Review by skidder — APPROVED** [review]
LGTM with some minor comments on whitespace that sneaked in
